### ISSUE TITLE
Platform restrictions on sub-dependencies are ignored

### ISSUE THREAD

**Issue by sudiamanj** [sdispater/poetry#700]
- [x] I am on the [latest](https://github.com/sdispater/poetry/releases/latest) Poetry version.
- [x] I have searched the [issues](https://github.com/sdispater/poetry/issues) of this repo and believe that this is not a duplicate.
- [x] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option).

- **OS version and name**: Windows 10
- **Poetry version**: 0.12.10
- **Link of a [Gist](https://gist.github.com/) with the contents of your pyproject.toml file**: https://gist.github.com/sudiamanj/83da2f18d435166a18479a1d87834128

## Issue

I am trying to add the [responder](https://github.com/kennethreitz/responder) dependency on Windows. Notice that their [setup.py](https://github.com/kennethreitz/responder/blob/v1.1.2/setup.py#L34) specifies that the `uvloop` dependency should *not* be included on Windows, as it is incompatible. However, it seems that Poetry is attempting to install it anyways:

```
PS Z:\sudiamanj\playground> poetry add responder
Using version ^1.1 for responder

Updating dependencies
Resolving dependencies... (0.3s)


Package operations: 19 installs, 0 updates, 0 removals

  - Installing uvloop (0.11.3)

[EnvCommandError]
Command ['C:\\Users\\sudiamanj\\AppData\\Local\\pypoetry\\Cache\\virtualenvs\\playground-py3.6\\Scripts\\pip.exe', 'install', '--no-deps', 'uvloop==0.11.3'] errored with the following output:
Collecting uvloop==0.11.3
  Using cached https://files.pythonhosted.org/packages/46/60/8e46188acba22369848d5e9bcbee3ca04b68a12d0598bd625c1a3994f5df/uvloop-0.11.3.tar.gz
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "C:\Users\SUDIAM~1\AppData\Local\Temp\pip-install-93rqsrxx\uvloop\setup.py", line 15, in <module>
        raise RuntimeError('uvloop does not support Windows at the moment')
    RuntimeError: uvloop does not support Windows at the moment

    ----------------------------------------
Command "python setup.py egg_info" failed with error code 1 in C:\Users\SUDIAM~1\AppData\Local\Temp\pip-install-93rqsrxx\uvloop\

add [-D|--dev] [--git GIT] [--path PATH] [-E|--extras EXTRAS] [--optional] [--python PYTHON] [--platform PLATFORM] [--allow-prereleases] [--dry-run] [--] <name> (<name>)...
```

**Comment by sdispater** [comment]
There is definitely something wrong here. I will investigate and get back to you.

**Comment by sdispater** [comment]
I investigated a bit and unfortunately there is nothing Poetry can do in this case.

While `responder` properly sets the appropriate environment marker (`sys_platform != 'win32'`), `uvicorn` which also depends on `uvloop` does not declare the appropriate marker but rather use if statements to declare dependencies programmatically by environment (https://github.com/encode/uvicorn/blob/master/setup.py#L43) which is not the recommended way to do so.

This an issue upstream in `uvicorn` and should be fixed there by following [PEP 508](https://www.python.org/dev/peps/pep-0508/#environment-markers).

**Comment by vlcinsky** [comment]
Just to keep track of related [uvicorn issue](https://github.com/encode/uvicorn/issues/219).

They seem to have rather fast development cycles so I hope to see a fix soon.

There is no need to reopen this issue as there is nothing poetry could do about this.

**Comment by vlcinsky** [comment]
Resolved in uvicorn v 0.4.6 by PR https://github.com/encode/uvicorn/pull/309

Uvicorn 0.5.* is likely to use extras (see https://github.com/encode/uvicorn/issues/219) to cover this issue so things can break or change again in some situations (hopefully get finally resolved).
